evap/evaluation/tests/tools.py

```diff
@@ -47,6 +50,30 @@ def let_user_vote_for_evaluation(app, user, evaluation):
     form.submit()
 
 
+def generate_fixtures(test_item):
+    """Decorator which annotates test methods as a fixture generation method.
+        The containing class is expected to include a `url` attribute which matches a valid path.
+        Unlike normal test methods, it should not assert anything and is expected to return a dictionary.
+        The key denotes the variant of the page to reflect multiple states, cases or views.
+        The value is a byte string of the html content."""
```

Comment:
Naming discussion! (yay?)
Alternatives for “page tests”:
- frontend tests
- end-to-end (or e2e) tests
- browser tests

Alternatives for “fixtures” as in “generate fixtures”:
- dummy pages
- page snapshots
- html test files

Polls are open and not limited to given choices :D

Reply:
I dislike fixtures a bit, and I think "page snapshots" or "rendered pages" makes it a lot clearer what we're talking about (keep in mind: django defines the term "fixture": https://docs.djangoproject.com/en/3.2/howto/initial-data/)

for "page tests", I think "frontend tests" is better, but I'd also be ok with "browser tests". In my mental model, I'd say we do two phases: page rendering and frontend testing -- so "e2e test" and "frontend test" would mean slightly different things here